spec/ruby/language/regexp_spec.rb

```diff
@@ -147,4 +147,24 @@
     pattern.should_not =~ 'fooF'
     pattern.should_not =~ 'T'
   end
+
+  ALL_TERMINATORS = (("!".."/").to_a + (":".."@").to_a + ("[".."`").to_a + ("{".."~").to_a)
+  SPECIAL_CASES = ['(', '{', '[', '<', '\\']
+  ESCAPABLE_TERMINATORS =  ['!', '"', '#', '%', '&', "'", ',', '-', ':', ';', '@', '_', '`']
+
+  ESCAPABLE_TERMINATORS.each do |c|
+    it "supports escaping #{c} when used as a terminator" do
+      ref = "(?-mix:#{c})"
+      pattern = eval ("%r" + c + "\\" + c + c)
+      pattern.to_s.should == ref
+    end
+  end
```

Comment:
ruby/spec avoids to define examples dynamically.
Could you move the `each` inside the `it` since it's overall specifying only one specific behavior?